src/dataloader.jl

```diff
@@ -1,75 +1,166 @@
-using StatsBase: sample, shuffle
-
-"""
-  stub -> <one of the dirs>/<img path> - _im/cr.tif
-"""
-function load_img(base::String, stub::String; rsize = (256,256))
-  im = joinpath(base, stub * "_im.tif")
-  cr = joinpath(base, stub * "_cr.tif")
-  x, y = load(im), load(cr)
-  x = imresize(x, rsize...)
-  y = imresize(y, rsize...)
-  x, y = channelview(x), channelview(y)
-  x = reshape(x, rsize..., 1, 1)
-  y = reshape(y, rsize..., 1, 1)
-  x, y
+
+@with_kw mutable struct Dataset
```

Comment:
This may not scale to how every dataset is stored, depending on the directory layout